backend/src/apiserver/storage/experiment_store.go

```diff
@@ -217,6 +236,116 @@ func (s *ExperimentStore) DeleteExperiment(id string) error {
 	return nil
 }
 
+func (s *ExperimentStore) ArchiveExperiment(expId string) error {
+	// ArchiveExperiment results in
+	// 1. The experiment getting archived
+	// 2. All the runs in the experiment getting archived no matter what previous storage state they are in
+	sql, args, err := sq.
+		Update("experiments").
+		SetMap(sq.Eq{
+			"StorageState": api.Experiment_STORAGESTATE_ARCHIVED.String(),
+		}).
+		Where(sq.Eq{"UUID": expId}).
+		ToSql()
+	if err != nil {
+		return util.NewInternalServerError(err,
+			"Failed to create query to archive experiment %s. error: '%v'", expId, err.Error())
+	}
+
+	updateRunsSql, updateRunsArgs, err := sq.
+		Update("run_details").
```

Comment:
Do we need to do anything for `jobs`?

Reply:
Thanks for catching this!
There is a little bit complexity here. We can sync quickly to be on the same page. 
First of all, Job doesn't have the status of being "archived". But it does have the status of being "enabled". Then, we have a couple of options 
(1) kind of abuse the "enabled" status in Job to simulate "unarchived" status in Run and Experiment, i.e., "enabled=true" means "unarchived" and "enabled=false" means "archived"
(2) add "archived" status for Job, but that would make the status of Job too complicated since it has two control factors now.

I will prefer (1) but want to check there will not introduce side effect. E.g. "enable" in Job has some existing specific meaning or treatment in KFP.

Reply:
(1) sounds good for me. I can't think of any side effect. @IronPan, anything catching your mind?

When an experiment is archived, I feel like disabling the job is the right action to take, otherwise the job is still scheduled to run, right? 
On the other side, when an experiment is unarchived, I think it's probably fine that we don't re-enable jobs that were previously disabled because of the experiment archiving. Because that requires extra complexity to keep track of why an job got disabled, which I don't think worth the effort.

Reply:
I agree with (1) and @chensun's arguments.

The behavior can be illustrated in a different way, we are not reusing `enabled` field as `archived`, but rather when we archive an experiment, it will automatically disable all jobs in that experiment. (very reasonable default behavior) Therefore, when we unarchive, it shouldn't enable all jobs.

Whether we need an separate archive field depends on if there are use-cases to archive an individual job. Sounds to me users can just delete jobs they don't want and disable jobs they want temporarily disabled. There's no need for an archive option.

Reply:
Used "enabled" field for job